omod-common/src/main/java/org/openmrs/module/webservices/helper/ServerLogActionWrapper.java

```diff
@@ -0,0 +1,90 @@
+/**
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.module.webservices.helper;
+
+import org.apache.log4j.Appender;
+import org.apache.log4j.Logger;
+import org.openmrs.api.context.Context;
+import org.openmrs.module.webservices.rest.web.RestConstants;
+import org.openmrs.util.MemoryAppender;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.regex.PatternSyntaxException;
+
+/***
+ * ServerLogActionWrapper used to serve the Server logs
+ */
+public class ServerLogActionWrapper {
+	
+	public List<String[]> serverLog;
+	
+	public void setServerLog(List<String[]> serverLog) {
+		this.serverLog = serverLog;
+	}
+	
+	public List<String[]> getServerLog() {
+		return serverLog;
+	}
+	
+	/***
+	 * Get server logs
+	 * 
+	 * @return List of last hundred server logs
+	 */
+	public List<String[]> getServerLogs() {
+		// Check the GET_SERVER_LOGS privilege to serve the server logs
+		Context.requirePrivilege(RestConstants.GET_SERVER_LOGS);
+		// Use the Memory Appender to retrieve the logs
+		Appender appender = Logger.getRootLogger().getAppender("MEMORY_APPENDER");
+		if (appender instanceof MemoryAppender) {
+			MemoryAppender memoryAppender = (MemoryAppender) appender;
+			List<String> logLines = memoryAppender.getLogLines();
+			List<String[]> finalOutput = new ArrayList<String[]>();
+			for (String logLine : logLines) {
+				String[] logElements = logLinePatternMatcher(logLine);
+				finalOutput.add(logElements);
+			}
+			return finalOutput;
+		}
+		else {
+			return new ArrayList<String[]>();
+		}
+	}
+	
+	/***
+	 * Match and find the patterns for log line
+	 * 
+	 * @param logLine Log lines from the terminal
+	 * @return Array of matched patterns
+	 */
+	public String[] logLinePatternMatcher(String logLine) {
+		String[] logElements = new String[4];
+		// Defined Pattern to analyze
+		String regExPatternType = "(INFO|ERROR|WARN|DEBUG)\\s.*?[-].*?\\s((?:[A-z][A-z].+))\\s[|](.*?)[|]\\s((.*\\n*)+)";
+		try {
+			Pattern pattern = Pattern.compile(regExPatternType);
+			Matcher matcher = pattern.matcher(logLine);
+			if (matcher.find()) {
+				// If pattern matches to the message
+				logElements[0] = matcher.group(1);
+				logElements[1] = matcher.group(2);
+				logElements[2] = matcher.group(3);
+				logElements[3] = matcher.group(4);
+			}
+			return logElements;
+		}
+		catch (PatternSyntaxException e) {
+			// In case of Exception, It will return empty array
```

Comment:
Then how will the client differentiate between finding no match and errors occurred?